src/Components/WebAssembly/Build/src/Tasks/MarkLazyLoadDependencies.cs

```diff
@@ -0,0 +1,136 @@
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using ResourceHashesByNameDictionary = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+{
+    public class MarkLazyLoadDependencies : Task
+    {
+        [Required]
+        public ITaskItem[] AssemblyReferences { get; set; }
+
+        [Required]
+        public ITaskItem[] PackageReferences { get; set; }
+
+        [Required]
+        public ITaskItem[] ExplicitlazyLoad { get; set; }
+
+        [Output]
+        public ITaskItem[] LazyLoadedAssemblies { get; set; }
+
+        private List<string> _visitedAssemblies = new List<string>();
+        private HashSet<ITaskItem> _lazyLoadAssemblies = new HashSet<ITaskItem>();
+
+        private const string LAZY_LOAD_LABEL = "BlazorLazyLoad";
+        private const string REFERENCE_SOURCE_KEY = "ReferenceSourceTarget";
+        private const string PROJECT_REFERENCE = "ProjectReference";
+        private const string ASSEMBLY_REFERENCE = "ResolveAssemblyReference";
+
+        public override bool Execute()
+        {
+            // Add assemblies that have been explicitly set by the user in a
+            // `BlazorLazyLoad` item group.
+            //
+            // <ItemGroup>
+            //  <BlazorLazyLoad Include="SomeBigAssembly.dll" />
+            // </ItemGroup>
+            _lazyLoadAssemblies.UnionWith(ExplicitlazyLoad);
+
+            foreach (var assembly in AssemblyReferences)
+            {
+                try
+                {
+                    ProcessAssembly(assembly);
+                }
+                catch (Exception e)
+                {
+                    Log.LogErrorFromException(e);
+                }
+            }
+
+            LazyLoadedAssemblies = _lazyLoadAssemblies.ToArray();
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private void ProcessAssembly(ITaskItem assembly)
+        {
+            var fileName = assembly.GetMetadata("FileName");
+
+            var visited = _visitedAssemblies.Contains(fileName);
+            if (visited)
+            {
+                return;
+            }
+
+            Log.LogMessage($"Checking if {fileName} should be marked for lazy-loading.");
+
+            _visitedAssemblies.Add(fileName);
+
+            // If the source is a project reference and the assembly is marked as
+            // lazy then we are processing a lazily-loaded project.
+            //
+            // <ProjectReference Include="SomeProj.csproj" BlazorLazyLoad="true" />
+            var sourceFromProjectReference = assembly.GetMetadata(REFERENCE_SOURCE_KEY) == PROJECT_REFERENCE;
+            if (sourceFromProjectReference && assembly.GetMetadata(LAZY_LOAD_LABEL) == "true")
+            {
+                Log.LogMessage($"{fileName} is a lazy-loaded project reference.");
```

Comment:
might want to log these with low improtance (I think it defaults to normal)

Reply:
Ended up removing the logs all together. They were helpful during development but not needed.